diff --git a/dist/index.cjs b/dist/index.cjs
index 074d2053454e491a4b3dadda253ab227434954f1..06da5e96e7d55fce04c2ad25f4477700fbb3c1b6 100644
--- a/dist/index.cjs
+++ b/dist/index.cjs
@@ -635,7 +635,7 @@ var betterFetch = async (url, options) => {
     const hasBody = context.method !== "HEAD";
     if (!hasBody) {
       return {
-        data: "",
+        data: response,
         error: null
       };
     }
diff --git a/dist/index.js b/dist/index.js
index 3fb614a9d607c3611f7a7631b7f541eba019b761..1c56068176f537162cc3ac84b5b328ee7221471e 100644
--- a/dist/index.js
+++ b/dist/index.js
@@ -588,7 +588,7 @@ var betterFetch = async (url, options) => {
     const hasBody = context.method !== "HEAD";
     if (!hasBody) {
       return {
-        data: "",
+        data: response,
         error: null
       };
     }
